client/src/routes/verify.js

```diff
@@ -1,39 +1,59 @@
-import { Component } from 'preact'
 import Error from './error'
 import config from '../../../config/client'
 import 'linkstate/polyfill'
-import withStyles from '../components/jss'
 
+import TokenPreview from '../components/tokenPreview'
 import { verify } from '../api/auth'
 import { route } from 'preact-router'
+import { useEffect, useState } from 'preact/hooks'
 
-export default withStyles({
-}, class Verify extends Component {
-  state = {
-    errors: {}
-  }
+function Verify () {
+  const [error, setError] = useState()
+  const [token, setToken] = useState()
 
-  componentDidMount () {
+  useEffect(() => {
     document.title = `Verify${config.ctfTitle}`
 
     const prefix = '#token='
     if (document.location.hash.startsWith(prefix)) {
-      route('/verify', true)
-
       const verifyToken = decodeURIComponent(document.location.hash.substring(prefix.length))
 
-      verify({ verifyToken })
-        .then(errors => {
-          this.setState({
-            errors
-          })
-        })
+      setToken(verifyToken)
     }
+
+    document.location.hash = ''
+  }, [])
+
+  const submitToken = () => {
```

Comment:
This needs to be in a `useCallback` to avoid creating a new function every render